modules/metrics/mule-metrics-api/src/main/java/org/mule/runtime/metrics/api/instrument/builder/LongCounterBuilder.java

```diff
@@ -31,6 +31,15 @@ public interface LongCounterBuilder {
    */
   LongCounterBuilder withUnit(String unit);
 
+  /**
+   * Sets the meter name for the instrument to build.
+   *
+   * @param meterName the meter name.
+   *
+   * @return the {@link LongCounterBuilder}.
+   */
+  LongCounterBuilder withMeterName(String meterName);
```

Comment:
I would not set the meter name as part of the instrument builder. The meter will provide the appropriate builder associated to itself. In anycase it can internally add its name if it is useful for an instrument to know what its meter is.